File: leetcode/1. Two Sum/solution1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        occured = {}
        
        for index, num in enumerate(nums):
            
            if target - num in occured:
                prev_index = occured[target - num]
                return [prev_index, index]
            
            occured[num] = index

File: leetcode/342. Power of Four/solution1.cpp
class Solution {
public:
    bool isPowerOfFour(int num) {
        if(num ==1)
        {
            return true;
        }
        if(num < 1){
            return false;
        }
        double x = log(num) / log(4);
        if((int)x == x) {
            return true;
        }
        return false;
        // int c = 0;
        // while(num%2 == 0) {
        //     c++;
        //     num /= 2;
        // }
        // return c && !(c%2);
    }
};
